valuable/build.rs

```diff
@@ -0,0 +1,41 @@
+#![warn(rust_2018_idioms, single_use_lifetimes)]
+
+use std::env;
+
+include!("no_atomic_cas.rs");
+
+// The rustc-cfg listed below are considered public API, but it is *unstable*
+// and outside of the normal semver guarantees:
+//
+// - `valuable_no_atomic_cas`
+//      Assume the target does *not* support atomic CAS operations.
+//      This is usually detected automatically by the build script, but you may
+//      need to enable it manually when building for custom targets or using
+//      non-cargo build systems that don't run the build script.
+//
+// With the exceptions mentioned above, the rustc-cfg strings below are
+// *not* public API. Please let us know by opening a GitHub issue if your build
+// environment requires some way to enable these cfgs other than by executing
+// our build script.
```

Comment:
In `futures`, there was a question about compatibility with custom targets, so we published a similar cfg as an unstable public API.

We can also treat it as a private API until there is an actual request, but this cfg is unstable and we can change/remove it at any time, so I think it is also ok as-is. (Whether this cfg is considered private or public depends on how the comments describe about it, so either way, it is a very simple change.)

Btw, one reason for the instability is that this cfg can eventually be replaced by `cfg(accessible)`. So I believe this cfg will never be a stable API.

Reply:
We'll need to implement similar detection logic for atomic integers later, so it's probably better to make this private for now. I'll update PR...